docs: edits to the dm join article (#1103)
cynkra/dm: vignettes/tech-dm-join.Rmd

First, we load the packages that we need:

> ```{r message=FALSE, warning=FALSE}

Author: To be consistent with all the other vignettes, this is taken care of in `setup.R`.

**Get all flights that can't be matched with airlines names.**

We expect the `flights` data from {nycflights13} package to be clean and well organized, so no flights should remain.
You can check this with an `anti_join`:

```{r}
dm %>%
  dm_flatten_to_tbl(flights, airlines, join = anti_join)
```

An example with filtering on a `dm` and then merging:

**Get all flights from Delta Air Lines which didn't depart from John F. Kennedy International Airport in May 2013 - and join all the airports data into the `flights` table.**

```{r}

> dm_nycflights13() %>%

Reviewer: We want the full data here.

```suggestion
dm_nycflights13(subset = FALSE) %>%
```

**Get all flights from Delta Air Lines which didn't depart from John F. Kennedy International Airport in May 2013 - and join all the airports data into the `flights` table.**

Currently, it is important to call `dm_apply_filters()` after piping your conditions.
Only then the underlying tables and key relations are updated and you can perform a join on the filtered data.
We are working towards removing this inconvenience [#62](https://github.com/cynkra/dm/issues/62).

```{r}
dm_nycflights13() %>%
  dm_filter(airlines, name == "Delta Air Lines Inc.") %>%
  dm_filter(flights, month == 5) %>%

>   dm_apply_filters() %>%

Author: AFAIK, no longer necessary.

Reviewer: This is about to be deprecated.

Reviewer: I meant that `dm_join_to_tbl()` will go away. Need to think about how to handle.

  dm_flatten_to_tbl(start = flights)
```

To be more precise, `dm_flatten_to_tbl()` will join all tables from one level of hierarchy (i.e., direct neighbors to table `start`).
If you want to cover tables from all levels of hierarchy, use `dm_squash_to_tbl()` instead.

Also, be aware that all column names need to be unique.
The `dm_flatten_to_tbl()` takes care of this by automatically renaming the relevant columns and informs the user if any names were changed, e.g. `dm_rename(airlines, airlines.name = name)`.

> If you want to merge all tables, but get a `dm` object in return, use `dm_wrap_tbl()` instead:
> 
> ```{r}
> dm_nycflights13() %>%
>   dm_wrap_tbl(root = flights)
> ```

Reviewer: ```suggestion
If you want to merge all tables, but get a nested table in return, use `dm_wrap_tbl()` with `pull_tbl()` instead:

```{r}
dm_nycflights13() %>%
  dm_wrap_tbl(root = flights) %>%
  pull_tbl(flights)
```
```

```

Be aware that all column names need to be unique.

> The `dm_flatten_to_tbl` takes care of this by automatically renaming the relevant columns and informs the user if any names were changed, e.g. `name -> airlines.name`.

Reviewer: New way of printing renames?

Reviewer 2: what does this comment mean, was it tackled @IndrajeetPatil?

Author: Yeah, I am not sure about this means either, so, no, it hasn't been tackled 😅

Reviewer 2: ah ok then let's wait for a clarification by @krlmlr :smiley_cat:

Author: @krlmlr I am still awaiting a clarification on this point. Thanks.

Reviewer: - The `dm_flatten_to_tbl()` function ...
- The format of the renaming messages `name -> airlines.name` has changed

> The `dm_flatten_to_tbl` takes cares of this by automatically renaming the relevant columns and informs the user if any names were changed, e.g. `name -> airlines.name`.

Reviewer: ```suggestion
The `dm_flatten_to_tbl` takes care of this by automatically renaming the relevant columns and informs the user if any names were changed, e.g. `name -> airlines.name`.
```

Also, the output of the renaming instructions have changed IIRC.
